components/layout/context.rs

```diff
@@ -120,31 +120,20 @@ impl<'a> LayoutContext<'a> {
         };
 
         // See if the image is already available
-        let result =
-            self.image_cache
-                .find_image_or_metadata(url.clone(), use_placeholder, can_request);
-        match result {
-            Ok(image_or_metadata) => Some(image_or_metadata),
-            // Image failed to load, so just return nothing
-            Err(ImageState::LoadError) => None,
-            // Not yet requested - request image or metadata from the cache
-            Err(ImageState::NotRequested(id)) => {
-                let image = PendingImage {
-                    state: PendingImageState::Unrequested(url),
-                    node: node.to_untrusted_node_address(),
-                    id: id,
-                };
-                self.pending_images
-                    .as_ref()
-                    .unwrap()
-                    .lock()
-                    .unwrap()
-                    .push(image);
-                None
-            },
+        if let Some(x) = self.image_cache.get_image(&url, use_placeholder) {
```

Comment:
Right, that makes sense. I guess I was thinking it might make it faster in cases where the image is available, but I realize now that the image is either not available or being tracked, in which case `get_image` is redundant